mrjob/py2.py

```diff
@@ -0,0 +1,170 @@
+"""Minimal utilities to make Python work for 2.6+ and 3.3+
+
+Strategies for making `mrjob` work across Python versions:
+
+Bytes vs. Unicode
+-----------------
+
+It's tempting to use `from __future__ import unicode_literals` and require
+that all non-byte strings be unicode. But that doesn't really make sense for
+Python 2, where str (bytes) and unicode can be used interchangeably.
+
+So really our string datatypes fall into two categories, bytes, and
+"strings", which are either ``str`` (i.e., bytes) or ``unicode`` in Python 2,
+and ``str`` (i.e. unicode) in Python 3.
+
+These things should always be bytes:
+
+- input data files
+  - use ``'b'`` when opening files: ``open(..., 'rb')``
+  - read data from ``sys.stdin.buffer`` in Python 3, not ``sys.stdin``
+- data encoded by protocols
+- data from subprocesses (this already happens by default)
+- log files parsed by mrjob
```

Comment:
Oh, I see, you're suggesting opening log files as unicode, in `internet` encoding. That *does* make sense; I agree that log files are meant to be unicode, not binary blobs.

Here's the issue: mrjob actually gets most of its log data through `cat()` in its "filesystem" interface, which is just a generator of `bytes` (not a filehandle). So it would be a fair amount of code in addition to Yelp's codec code to make that work. I can't just wrap it in `codecs.getreader()`.

So I'm going to just use `from_bytes()` instead. I *will* add cp1252 encoding (see below).

Also, will probably want to use `yelp_encoding` for my startup's code; don't mind extra deps there.

Did you notice that the docstring for [internet.py](https://github.com/Yelp/yelp_encodings/blob/master/yelp_encodings/internet.py) is incorrect? It doesn't mention cp1252, and I'm not sure what the remark about "latin1html5" means.